### DISCUSSION TITLE
Questions about 'PgSTACReader' and 'SimpleSTACReader'

### DISCUSSION THREAD

**Discussion by louisstuart96** [stac-utils/titiler-pgstac#212] [Q&A] [answered]
As far as I have found, 'Searches' and 'Collections' endpoints are built by `titiler.pgstac.factory.MosaicTilerFactory`, which uses `titiler.pgstac.reader.SimpleSTACReader`; 'Items' endpoints are built by `titiler.core.factory.MultiBaseTilerFactory`, which uses `titiler.pgstac.reader.PgSTACReader` (inherited from `rio_tiler.io.stac.STACReader`).

What are the differences between these two readers? Our team is trying to make some customizations, like support of HTTP and S3 links in parallel, and handling 'alterate_asset' STAC extensions. Do we need to customize both readers?

**Comment by vincentsarago** [comment, marked as answer]
- `rio_tiler.io.stac.STACReader` expects either a Path or a Pystac Item (internally it will fetch the stac Item if only a path is passed)

- `titiler.pgstac.reader.PgSTACReader`, used in `/items` expects a single `input` as a pystac Item 

- `titiler.pgstac.reader.SimpleSTACReader` expects a `Dict` as input, from pgstac response directly 

You'll need to customize each ones but some `customization` around alternate href is already available https://github.com/cogeotiff/rio-tiler/blob/main/rio_tiler/io/stac.py#L348-L350

I can't remind why it's not done the same in titiler-pgstac but we could certainly bring this into https://github.com/stac-utils/titiler-pgstac/blob/d0d2f868801f3c47b0a34f70472687aec3a32cbd/titiler/pgstac/reader.py#L151-L155
